core/src/main/java/nl/nn/adapterframework/jta/narayana/NarayanaDataSourceFactory.java

```diff
@@ -29,21 +29,31 @@
 import com.arjuna.ats.internal.jdbc.drivers.modifiers.ModifierFactory;
 import com.arjuna.ats.jta.recovery.XAResourceRecoveryHelper;
 
+import lombok.Getter;
 import lombok.Setter;
 import nl.nn.adapterframework.jndi.JndiDataSourceFactory;
+import nl.nn.adapterframework.util.AppConstants;
 import nl.nn.adapterframework.util.LogUtil;
 
 public class NarayanaDataSourceFactory extends JndiDataSourceFactory {
 	protected static Logger log = LogUtil.getLogger(NarayanaDataSourceFactory.class);
 
+	private @Getter @Setter int maxPoolSize=20;
+
 	private @Setter NarayanaJtaTransactionManager transactionManager;
 
+	public NarayanaDataSourceFactory() {
+		AppConstants appConstants = AppConstants.getInstance();
+		maxPoolSize = appConstants.getInt(MIN_POOL_SIZE_PROPERTY, maxPoolSize);
```

Comment:
Is this intentional, or a typo?
If intentional, it would be good to explain in a comment.

If a typo, can we please have unit tests around these things, with the actual implementations and not just an IAF-test with some mocks? Might help catch typos like this.